content/blaze.md

```diff
@@ -0,0 +1,543 @@
+      // Handle for launch screen defined in app-body.js
+      AppLaunchScreen.listRender.release();
+    }
+  });
+});
+```
+
+<h2 id="smart-components">Writing smart components with Blaze</h2>
+If your component needs to access state outside of its data context---for instance, data from the server via subscriptions or the the contents of client-side store, then you should be careful how you do that accessing. As discussed in the [data loading article](data-loading) you should be careful and considered in how use such smart components.
+
+To begin with, all of the rules of thumb about reusable components apply to smart components. In addition:
+
+<h3 id="subscribing">Subscribe from `onCreated`</h3>
+You should subscribe to publications on the server from an `onCreated` callback (within an `autorun` block if you are reliant on reactively changing arguments). In the Todos example app, in the `listsShowPage` template we subscribe to the `list/todos` publication based on the current `_id` FlowRouter param:
+
+```js
+Template.listsShowPage.onCreated(function() {
+  this.state = new ReactiveDict();
+  this.autorun(() => {
+    const listId = FlowRouter.getParam('_id');
+    this.state.set({listId});
+    this.subscribe('list/todos', listId);
+  });
+});
+```
+
+By using `this.subscribe()` (as opposed to `Meteor.subscribe`), the subscription state automatically gets amalagamated into the template instance's subscription readiness reactive state variable, which can be used both from within templates (via the `subscriptionsReady` helper) or within helpers (via `Template.subscriptionsReady()`).
+
+Notice as well in this case that we access the global client-side state store `FlowRouter` in the `onCreated()` callback also, storing its value locally in a local `state` dictionary.
+
+XXX: should instead we just read from flow router a second time here: https://github.com/meteor/todos/blob/master/packages/lists-show/lists-show-page.js#L28 ??
+
+<h3 id="fetch-in-smart-components">Fetch in helpers</h3>
+Typically, as [outlined in the ui/ux article](ui-ux.md#smart-components) you should fetch data in the same component that you subscribe in. In Blaze, it's usually simplest to fetch the data in a helper, which you can then use to pass data into a reusable child component. For example, in the `listShowPage`: 
+
+```html
+{{> listsShow todosReady=Template.subscriptionsReady
+  list=(getFullList listIdOnly) todos=listIdOnly.todos}}
+```
+
+<h2 id="reusing-code">Reusing code in Blaze</h2>
+It's not uncommon to want to reuse code between two otherwise unrelated components. There are two principal ways to do this in Blaze.
+
+<h3 id="composition">Composition</h3>
+If possible, it's usually best to try and abstract out the reusable part of the two components that need to share functionality. If you are used to following patterns to [create reusable components](#reusable-components), then it should be simple to reuse those components in many places. 
```

Comment:
It doesn't look like you pushed this change. After this we can merge! Thanks for doing this article, must have been a slog